src/include/duckdb/common/types.hpp

```diff
@@ -12,6 +12,7 @@
 #include "duckdb/common/constants.hpp"
 #include "duckdb/common/single_thread_ptr.hpp"
 #include "duckdb/common/vector.hpp"
+#include "duckdb/common/windows.hpp"
```

Comment:
The `types.hpp` header gets included in almost every single file of the system - and `windows.h` is not particularly light on Windows. Can't we move `infinity` and `ninfinity` functions out of the header (as non-inline functions) and have them use `NumericLimits` instead?